frontend/src/pushapi.js

```diff
@@ -102,92 +51,65 @@ angular.module('clouway-push', [])
       /**
        * Open channel using specified channel token.
        *
-       * @param {String} channelToken the token to use for opening channel.
+       * @param {String} data the data from server to use for opening channel.
        */
-      var openChannel = function (channelToken) {
-        var channel = new goog.appengine.Channel(channelToken);
-        var socket = channel.open();
-
-        socket.onmessage = onMessage;
-
-        socket.onerror = function (errorMessage) {
-          establishConnection();
-        };
+      var openChannel = function (data) {
+        return firebase.auth().signInWithCustomToken(data.token).then(function () {
+          customerName = data.customerName;
+        }, function (error) {
+        });
       };
 
       var subscribeForEvents = function (events) {
-        var subscriber = service.openConnection();
-        var params = {
-          subscriber: subscriber,
-          eventName: events
-        };
-
         // Used for testing purposes only.
         if (!endpoints.serviceUrl || endpoints.serviceUrl === '') {
           return;
         }
 
         if (events.length) {
-          $http.put(endpoints.serviceUrl, '', {params: params}).then(function (data) {
-
+          service.openConnection().then(function () {
+            firebaseDatabaseRef = firebase.database().ref(customerName);
+            angular.forEach(events, function (event) {
+              var fbDbChildRef = firebaseDatabaseRef.child(event.eventName);
+              var isInitialBind = true;
+
+              //to skip initial binding of added child
+              fbDbChildRef.once('value', function () {
+                isInitialBind = false;
+              });
+
+              fbDbChildRef.on('child_added', function child_added(result) {
```

Comment:
No need to have these callbacks named - they can just be anonymous.

Reply:
Done. Now callbacks are anonymous.